src/plugins/intel_cpu/src/emitters/snippets/x64/jit_loop_emitters.cpp

```diff
@@ -133,118 +111,72 @@ ov::snippets::lowered::ExpressionPtr jit_loop_end_emitter::get_loop_begin_expr(c
 }
 
-void jit_loop_end_static_emitter::emit_impl(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
+size_t jit_loop_end_emitter::aux_gprs_count() const {
+    return need_runtime_args ? 1 : 0;
+}
+
+void jit_loop_end_emitter::emit_impl(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
     std::vector<size_t> data_ptr_reg_idxs;
     // the last input is actually a work_amount reg
-    data_ptr_reg_idxs.reserve(num_inputs - 1);
+    data_ptr_reg_idxs.reserve(num_inputs + num_outputs);
     std::copy(in.begin(), in.end() - 1, std::back_inserter(data_ptr_reg_idxs));
 
+    const auto id_offset = loop_id * sizeof(jit_snippets_call_args::loop_args_t);
+    Reg64 reg_increments = need_runtime_args ? Reg64(static_cast<int>(aux_gpr_idxs[0])) : Reg64();
     Reg64 reg_work_amount = Reg64(in.back());
-    if (!evaluate_once) {
-        for (size_t idx = 0; idx < data_ptr_reg_idxs.size(); idx++) {
-            if (!is_incremented[idx] || ptr_increments[idx] == 0)
-                continue;
-            Reg64 data_reg = Reg64(static_cast<int>(data_ptr_reg_idxs[idx]));
-            h->add(data_reg, ptr_increments[idx] * wa_increment * data_sizes[idx]);
-        }
-        h->sub(reg_work_amount, wa_increment);
-        h->cmp(reg_work_amount, wa_increment);
-        h->jge(*loop_begin_label);
-    }
 
-    for (size_t idx = 0; idx < data_ptr_reg_idxs.size(); idx++) {
-        if (!is_incremented[idx] || finalization_offsets[idx] == 0)
-            continue;
-        Reg64 data_reg = Reg64(static_cast<int>(data_ptr_reg_idxs[idx]));
-        h->add(data_reg, finalization_offsets[idx] * data_sizes[idx]);
+    auto apply_increment = [&](size_t idx, int64_t increment, int64_t scale = 1) {
+        if (is_incremented[idx] && increment != 0) {
+            if (ov::snippets::utils::is_dynamic_value(increment)) {
+                h->add(Reg64(static_cast<int>(data_ptr_reg_idxs[idx])), h->ptr[reg_increments + idx * sizeof(int64_t)]);
+            } else {
+                h->add(Reg64(static_cast<int>(data_ptr_reg_idxs[idx])), increment * scale * data_sizes[idx]);
+            }
+        }
+    };
+
+#define APPLY_INCREMENTS(COND, TYPE, INCREMENTS, SCALE)                                         \
+    if (COND) {                                                                                 \
+        Reg64 reg_runtime_params = abi_param1; /* defined by jit_kernel_emitter */              \
+        h->mov(reg_increments, h->ptr[reg_runtime_params + GET_OFF(loop_args)]);                \
+        h->mov(reg_increments, h->ptr[reg_increments + id_offset + GET_OFF_LOOP_ARGS(TYPE)]);   \
+    }                                                                                           \
+    for (size_t idx = 0; idx < data_ptr_reg_idxs.size(); idx++) {                               \
+        apply_increment(idx, INCREMENTS[idx], SCALE);                                           \
     }
```

Comment:
1. Why not, done
2. Because firstly we should store `loop_args` to `reg_increments`. Secondly we should store data pointer shift parameters from `reg_increments` (it contains `loop_args`) and push in the same register `reg_increments` to avoid using one more aux gpr (we can only one gpr aux in loop emitters: 11 data pointers, 2 work_amount, 1 RSP, 1 RBP = used 15 gprs)
3. Good idea, thanks!